### PR TITLE
Helm 3

### LINKED ISSUE #56: Upgrade chart to Support Helm 3
With the release of [Helm 3] and v2 of the API, we should look at migrating the chart to the new version. ~Looking to get an idea if people are ready to make that change and/or if there's a reason to hold off on migrating the chart yet. #53 already does this migration so taking it in now would resolve this issue. We want to get an idea if this is something users are okay with.~ **The only negative thing about this change is that you can't use the Helm 2 CLI with the chart once it goes to v2 of the API.**

EDIT: We're pretty far into the lifecyle of Helm 3 now and Helm 2 is only has three months of security patches left. We should cut over now. https://helm.sh/blog/helm-v2-deprecation-timeline/

Notes:
- https://helm.sh/blog/helm-3-released/
- Helm v2 deprecation timeline (6 mo bug/1 yr security): https://helm.sh/blog/2019-10-22-helm-2150-released/#helm-2-support-plan
- Migration tooling: https://helm.sh/blog/migrate-from-helm-v2-to-helm-v3/
- Check out PR #53 for what needs to be updated to be fully helm 3 compatible

[Helm 3]:https://helm.sh/blog/helm-3-released/

I'm unsure if any issues currently do exist when trying to use helm 3 with the current version of the chart.

---
## Goal

Using Helm v3 with the chart works without any issues.

## Tasks

Updating the chart first is probably a good first step. Then let the jobs that fail in CI guide the rest of this story/epic.

- [x] Make necessary changes to concourse-chart so it works with helm 3 (see #53 for pointers)
  - [x] Update `README` to ensure all commands shown work with helm 3
- [x] Update k8s-topgun to run with `helm` v3. In-progress: https://github.com/concourse/concourse/tree/helm3-k8s-topgun
  - concourse/concourse#5219
- [ ] k8s-topgun tests for external postgres and Prometheus should be updated to not use the deprecated `helm/charts` repo 
- [x] Update CI to use helm v3 (docker images, task scripts, etc.)
- [x] Update the [enterprise helm pipeline](https://ci.concourse-ci.org/teams/main/pipelines/enterprise-helm-chart)
  - [x] Package the Helm 3 CLI in the `helm-cli` job
  - [ ]  Update install docs: https://docs.pivotal.io/p-concourse/v6/installation/install-concourse-helm/
- [ ] Migrate our clusters to helm3 using the plugin [2to3](https://github.com/helm/helm-2to3)
  - [ ] hush-house
  - [ ] cluster-1
  - [ ] greenpeace
  - [ ] topgun

### PR DESCRIPTION
**Pull request by xtreme-vikram-yadav**
# Existing Issue

Fixes #56  .

# Changes proposed in this pull request

* Update charts to support helm3

# Contributor Checklist

- [x] Variables are documented in the `README.md`
- [x] Which branch are you merging into?
    - `dev` is for changes related to the next release of Concourse (aka unpublished code on `master` in [concourse/concourse](https://github.com/concourse/concourse))

# Reviewer Checklist
> This section is intended for the core maintainers only, to track review progress. Please do not
> fill out this section.
- [ ] Code reviewed
- [ ] Topgun tests run
- [ ] Back-port if needed
- [ ] Is the correct branch targeted? (`master` or `dev`)

### CODE REVIEW

File: requirements.lock [hunk 1]
```diff
@@ -1,6 +0,0 @@
-dependencies:
-- name: postgresql
-  repository: https://charts.bitnami.com/bitnami
-  version: 9.2.0
-digest: sha256:5082c9ee31559622b64a7391ad9d2f8cf8488a0d34ff03fb1600dfac5f3d89e2
-generated: "2020-08-28T15:21:20.204476-04:00"
```
Comment (chenbh) [inline requirements.lock:6]: The [helm documentation](https://helm.sh/docs/faq/#consolidation-of-requirementsyaml-into-chartyaml) mentioned that `requirements.lock` was consolidated into `Chart.lock`, is that missing from this PR?

**Comment by supahgreg** [comment]
Minor, but the `helm install` near the start of the README should be updated to include a release name (or `--generate-name`) since v3 no longer auto-generates one.

File: requirements.lock (hunk 1 above)
Comment (xtreme-vikram-yadav) [reply requirements.lock:6]: Indeed. We definitely missed committing that.

**Review by taylorsilva — APPROVED** [review]

**Comment by taylorsilva** [comment]
Can't publish a new major of the chart until concourse/ci#375 is merged because that updates all of our scripts in CI to use helm v3

**Pull request merged**